### ISSUE TITLE
Constructing query for 4 field search

### ISSUE THREAD

**Issue by iteles** [FAC-GM/app#166]
Linked to #134.

Constructing the queries for the four individual search fields:
+ Skill - Look for this first in the job title, following by skills list, followed by rest of the profile
+ Location
+ Job title
+ Name

**Comment by SimonLab** [comment]
I think the summary of the profile is importatnt too

- Skill - Look for this first in the job title, following by skills list, **summuray** and followed by rest of the profile

**Comment by iteles** [comment]
How about _past_ job titles?
I think we need a list of the priority of places where we search for the skills :+1:

**Comment by nelsonic** [comment]
@iteles *past* job titles would be an *interesting* thing to *test*. 
In many cases developers are no longer *interested* in doing the "old" thing...

**Comment by iteles** [comment]
Good point :+1:

**Comment by SimonLab** [comment]
To clarify, do we want to be able to search for a list of skills or just for one skill?

**Comment by nelsonic** [comment]
List. @iteles suggested we ask them to separate each skill with a comma.

We had discussed maintaining a list of the skills so we could have auto-completion.
But that would be over-kill at this stage. Just allowing multiple skills with commas between them *should* be adequate.

**Comment by SimonLab** [comment]
We have now manage to create a request targeting location and 
body: {
  query: {
    bool: {
      must: [
         {
            match: {
              location: request.query.location
            }
         },
         {
           match: {
              fullname: request.query.name
           }
         }
        ]
      }
    }
  }
```
We have however some issue with some special character. The next step is to verify which analyser we are using when we index documents and which analyser we use for a search. How to check which analyser we are using?

**Comment by heron2014** [comment]
Also on top of @SimonLab comment we were investigating difference between **term** and **match**. From our observation we come to conclusion that whenever you use **term** - it will query exact values as they are in database and **match** is more relaxed, it also matches lowercase and uppercase cases.

**Comment by nelsonic** [comment]
@SimonLab this query looks good, however I would construct it *dynamically*. Let me know if you want to pair...

**Comment by SimonLab** [comment]
We have are doing dynamically [here](https://github.com/FAC-GM/app/blob/fe2b732a780f3e8131b7the725296fda66edb31c535/lib/search_query.js#L38-L60). The request is working but the handler suffer from the pagination functionality. I'm looking now at how to simplify the pagination.

**Comment by nelsonic** [comment]
@SimonLab please create a PR with the branch so you can keep track of the build.
at the moment it looks like you need to add some config to ensure the tests pass. see: https://travis-ci.org/FAC-GM/app/builds/96084707

**Comment by iteles** [comment]
:+1: Merged in with PR #181 .
